lib/compress/zstd_compress.c

```diff
@@ -2491,15 +2496,22 @@ static size_t ZSTD_compressBlock_targetCBlockSize(ZSTD_CCtx* zc,
                     srcSize,
                     zc->entropyWorkspace, HUF_WORKSPACE_SIZE /* statically allocated in resetCCtx */,
                     zc->bmi2);
-            if (!ZSTD_isError(cSize) && cSize != 0) {
+
+            FORWARD_IF_ERROR(cSize);
+            if (cSize == 0) {
+                /* If compressSequences didn't work, we just output a regular
+                 * uncompressed block */
+                cSize = ZSTD_noCompressBlock(dst, dstCapacity, src, srcSize, lastBlock);
+                FORWARD_IF_ERROR(cSize);
+            } else {
                 U32 const cBlockHeader24 = lastBlock + (((U32)bt_compressed)<<1) + (U32)(cSize << 3);
                 MEM_writeLE24(ostart, cBlockHeader24);
                 cSize += ZSTD_blockHeaderSize;
```

Comment:
Do we need to set `noCompressSuperBlock = 0` in this branch?

Reply:
Assuming that is correct, can we add a test that catches this bug, since it indicates lacking test coverage.

I would be okay if you can show that `fuzzer.c`, `zstreamtest.c`, or any of the libfuzzer fuzzers can find the bug within 1 hour. I just want to know that if we landed this change as-is, we would be alerted of the bug before we cut a release. Ideally we would know before landing.